House-of-glass/src/Contexts/Cart/CartContext.jsx

```diff
@@ -0,0 +1,174 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import axios from "axios";
+import { cartReducer } from "./CartReducer";
+import Toast from "components/Toast/Toast";
+
+const CartContext = createContext();
+
+const useCart = () => useContext(CartContext);
+
+const CartProvider = ({ children }) => {
+  const initialCartState = {
+    cart: [],
+    cartCount: 0,
+    cartSum: 0,
+    cartDiscount: 0,
+  };
+  const [cartState, cartDispatch] = useReducer(cartReducer, initialCartState);
+
+  const token = localStorage.getItem("Token");
+  const header = {
+    headers: {
+      authorization: token,
+    },
+  };
```

Comment:
This part gave me trouble after I deployed to vercel. Steps you might have to follow to reproduce the error

1, Open the deployed link
2. Login
3. Do some functionality such as add to cart - gives 500 error

Reason for this:  

header value is not getting called at run time. Once the component is mounted, it is not being re run after you login as there is change in the component

```
const header = {
 headers: {
      authorization: token,
    },
  };
```

Solution: supply the headers normally.

Reply:
Thanks for the heads-up on this. Have refactored the code accordingly.